File: app/src/main/java/com/mojiscan/ocr/network/ApiService.java
package com.mojiscan.ocr.network;

import com.mojiscan.ocr.data.model.ErrorResponse;
import com.mojiscan.ocr.data.model.HealthResponse;
import com.mojiscan.ocr.data.model.OCRResponse;
import com.mojiscan.ocr.data.model.ProcessResponse;
import com.mojiscan.ocr.data.model.TranscribeResponse;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;

public interface ApiService {
    @Multipart
    @POST("api/v1/process")
    Call<ProcessResponse> processFile(
            @Part MultipartBody.Part file,
            @Part("process_type") RequestBody processType
    );

    @Multipart
    @POST("api/v1/ocr")
    Call<OCRResponse> processOCR(
            @Part MultipartBody.Part file
    );

    @Multipart
    @POST("api/v1/transcribe")
    Call<TranscribeResponse> processTranscribe(
            @Part MultipartBody.Part file,
            @Part("language") RequestBody language,
            @Part("model") RequestBody model
    );

    @GET("api/health")
    Call<HealthResponse> healthCheck();
}

File: app/src/main/java/com/mojiscan/ocr/network/ApiService.kt
package com.mojiscan.ocr.network

import com.mojiscan.ocr.data.model.*
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    @Multipart
    @POST("api/v1/process")
    suspend fun processFile(
        @Part file: MultipartBody.Part,
        @Part("process_type") processType: RequestBody?
    ): Response<ProcessResponse>

    @Multipart
    @POST("api/v1/ocr")
    suspend fun processOCR(
        @Part file: MultipartBody.Part
    ): Response<OCRResponse>

    @Multipart
    @POST("api/v1/transcribe")
    suspend fun processTranscribe(
        @Part file: MultipartBody.Part,
        @Part("language") language: RequestBody?,
        @Part("model") model: RequestBody?
    ): Response<TranscribeResponse>

    @GET("api/health")
    suspend fun healthCheck(): Response<HealthResponse>
}

